core/foundation/test/testRRangeCast.cxx

```diff
@@ -0,0 +1,21 @@
+#include "ROOT/RRangeCast.hxx"
+
+#include "gtest/gtest.h"
+
+class ClassA {
+};
+class ClassB : public ClassA {
+};
+
+TEST(RRangeCast, CStyleArray)
+{
+
+   ClassB b;
+   const ClassA *arr[3] = {&b, &b, &b};
+   int n = 0;
+   for (auto *item : ROOT::RangeStaticCast<const ClassB *>(arr)) {
+      EXPECT_TRUE(item != nullptr);
```

Comment:
Maybe even
```suggestion
      EXPECT_EQ(item, &b);
```

Reply:
👍 
and does something along the lines of the following make sense? The `auto` hides everything, so you might statically ensure that you get what you want.
```c++
static_assert(std::is_same<decltype(*item),ClassB>::value, ...);
```

Same for `dynamic`?

Reply:
Okay! I now consider also the type in `EXPECT_EQ` by doing:
```C++
EXPECT_EQ(item, static_cast<const ClassB *>(arr[n]));
```